### ISSUE TITLE
Project Vision

### ISSUE THREAD

**Issue by isiahto** [isiahto/Jpegdecoder#8]
After #7 is done, the metadata reading part is pretty much completed (minus the SOS marker node), and here is the vision for this project:

### Finish the Goal First
- [X] Read all the markers
- [ ] Decoding process (Big Item)

### Ready-to-ship Goal (would be a new story for each)
- [ ] A processor to automatically read the markers (as oppose to now which is hardcoded)
- [ ] Add references (or enum) or some specific values (i.e. DHT class: 0=DC, 1=AC)
- [ ] Refactor some of the naming of the variables/methods
